### PR TITLE
Fix IPNetwork inventory

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

**Comment by steph461** [comment]
@cedric-anne 

https://github.com/glpi-project/glpi/blob/1212e127a6fb8c7a311e6b20a2848781908efd91/src/Inventory/Asset/InventoryNetworkPort.php  only ?

this is for this issue #10960 ?

**Comment by cedric-anne** [comment]
> https://github.com/glpi-project/glpi/blob/1212e127a6fb8c7a311e6b20a2848781908efd91/src/Inventory/Asset/InventoryNetworkPort.php only ?

yes

> this is for this issue #10960 ?

No idea, I just came across this piece of code by accident. I do not really know how network inventory works, by I am pretty sure that this code was wrong.

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 1]
```diff
@@ -1436,13 +1436,13 @@ public function testUpdateComputer()
...
         ]);
-        $this->integer(count($logs))->isIdenticalTo(72);
+        $this->integer(count($logs))->isIdenticalTo(74);
 
         $expected_types_count = [
             \Log::HISTORY_CREATE_ITEM => 64,
             \Log::HISTORY_ADD_SUBITEM => count($expecteds_fs),
             0 => 1, // Change Monitor contact (is_contact_autoupdate)
-            \Log::HISTORY_ADD_RELATION => 4 //OS and Monitor on both sides
+            \Log::HISTORY_ADD_RELATION => 6 // 2 IPNetwork/IPAdresss + OS and Monitor on both sides
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:1445]: Now, 2 IPNetwork/IPAddress items are created.

First one correspond to 
```json
            {
                "description": "enp0s25",
                "driver": "e1000e",
                "ipaddress": "172.28.220.96",
                "ipgateway": "172.28.220.1",
                "ipmask": "255.255.255.0",
                "ipsubnet": "172.28.220.0",
                "mac": "b8:6b:23:e0:5c:81",
                "pciid": "8086:1559:1179:0001",
                "pcislot": "0000:00:19.0",
                "speed": "100",
                "status": "up",
                "type": "ethernet",
                "virtualdev": false
            },
```

Second one corresponds to
```json
            {
                "description": "wlp2s0",
                "driver": "iwlwifi",
                "ipaddress": "172.28.220.181",
                "ipgateway": "172.28.220.1",
                "ipmask": "255.255.255.0",
                "ipsubnet": "172.28.220.0",
                "mac": "30:3a:64:ee:65:88",
                "pciid": "8086:08B3:8086:8170",
                "pcislot": "0000:02:00.0",
                "status": "up",
                "type": "wifi",
                "virtualdev": false,
                "wifi_bssid": "00:1D:7E:43:96:59",
                "wifi_mode": "Managed",
                "wifi_ssid": "DaWifi",
                "wifi_version": "802.11"
            },
```

See https://raw.githubusercontent.com/glpi-project/inventory_format/master/examples/computer_3.json

File: tests/functionnal/Glpi/Inventory/Inventory.php [hunk 2]
```diff
@@ -1779,11 +1779,12 @@ public function testUpdateComputer()
             'OFFSET' => $nblogsnow,
         ]);
 
-        $this->integer(count($logs))->isIdenticalTo(45);
+        $this->integer(count($logs))->isIdenticalTo(47);
 
         $expected_types_count = [
             \Log::HISTORY_DELETE_SUBITEM => 5,//networkport and networkname
-            \Log::HISTORY_CREATE_ITEM => 16, //virtual machines, os, manufacturer, net ports, net names, ...
+            \Log::HISTORY_ADD_RELATION => 1, //new IPNetwork/IPAddress
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Inventory.php:1786]: Now, on update, a new IPNetwork/IPAddress item is created.

It corresponds to:
```json
            {
                "description": "enp0s25",
                "driver": "e1000e",
                "ipaddress": "192.168.1.103",
                "ipgateway": "192.168.1.1",
                "ipmask": "255.255.255.0",
                "ipsubnet": "192.168.1.0",
                "mac": "b8:6b:23:e0:5c:81",
                "pciid": "8086:1559:1179:0001",
                "pcislot": "0000:00:19.0",
                "speed": "100",
                "status": "up",
                "type": "ethernet",
                "virtualdev": false
            },
```

See https://raw.githubusercontent.com/glpi-project/inventory_format/master/examples/computer_3_updated.json

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**